code/modules/vehicles/motorbike.dm

```diff
@@ -0,0 +1,137 @@
+
+/obj/vehicle/ridden/motorbike/examine(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	to_chat(user, "To access internal storage click with an empty hand or drag the bike onto self.")
+	to_chat(user, "The fuel gauge on the bike reads \"[fuel_count/fuel_max*100]%\"")
+
+/obj/vehicle/ridden/motorbike/post_buckle_mob(mob/living/M)
+	add_overlay(motorbike_cover)
+	return ..()
+
+/obj/vehicle/ridden/motorbike/post_unbuckle_mob(mob/living/M)
+	if(!LAZYLEN(buckled_mobs))
+		cut_overlay(motorbike_cover)
+	return ..()
+
+/obj/vehicle/ridden/motorbike/welder_act(mob/living/user, obj/item/I)
+	if(obj_integrity >= max_integrity)
+		return TRUE
+	balloon_alert_to_viewers("[user] starts repairs", ignored_mobs = user)
+	balloon_alert(user, "You start repair")
+	if(!do_after(user, 2 SECONDS))
+		balloon_alert_to_viewers("Stops repair")
+		return
+	if(!I.use_tool(src, user, 0, volume=50, amount=1))
+		return TRUE
+	obj_integrity += min(10, max_integrity-obj_integrity)
+	if(obj_integrity == max_integrity)
+		balloon_alert_to_viewers("Fully repaired!")
+	else
+		balloon_alert_to_viewers("[user] repairs", ignored_mobs = user)
+		balloon_alert(user, "You repair damage")
+	return TRUE
+
+/obj/vehicle/ridden/motorbike/attack_hand(mob/living/user)
+	return motor_pack.open(user)
+
+/obj/vehicle/ridden/motorbike/MouseDrop(obj/over_object)
+	if(motor_pack.handle_mousedrop(usr, over_object))
+		return ..()
+
+/obj/vehicle/ridden/motorbike/Move(direction)
+	. = ..()
+	if(!.)
+		return
+	if(!LAZYLEN(buckled_mobs)) // dont use fuel or make noise unless we're being used
+		return
+	fuel_count--
+
+	if(COOLDOWN_CHECK(src, enginesound_cooldown))
+		COOLDOWN_START(src, enginesound_cooldown, 20)
+		playsound(get_turf(src), 'sound/vehicles/carrev.ogg', 100, TRUE)
+
+/obj/vehicle/ridden/motorbike/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(istype(I, /obj/item/reagent_containers/jerrycan))
+		var/obj/item/reagent_containers/jerrycan/gascan = I
+		if(gascan.reagents.total_volume == 0)
+			balloon_alert(user, "Out of fuel!")
+			return
+		if(fuel_count >= fuel_max)
+			balloon_alert(user, "Already full!")
+			return
+
+		var/fuel_transfer_amount = min(gascan.fuel_usage*2, gascan.reagents.total_volume)
+		gascan.reagents.remove_reagent(/datum/reagent/fuel, fuel_transfer_amount)
+		fuel_count = min(fuel_count + FUEL_PER_CAN_POUR, fuel_max)
+		playsound(loc, 'sound/effects/refill.ogg', 25, 1, 3)
+		balloon_alert(user, "[fuel_count/fuel_max*100]%")
+		return TRUE
+	if(user.a_intent != INTENT_HARM)
+		return motor_pack.attackby(I, user, params)
+
+/obj/vehicle/ridden/motorbike/obj_break()
+	START_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/vehicle/ridden/motorbike/process()
+	if(obj_integrity >= integrity_failure * max_integrity)
+		return PROCESS_KILL
+	if(prob(20))
+		return
+	var/datum/effect_system/smoke_spread/smoke = new
+	smoke.set_up(0, src)
+	smoke.start()
+
+/obj/vehicle/ridden/motorbike/bullet_act(obj/projectile/P)
+	if(prob(50) || !buckled_mobs)
```

Comment:
50% change of dodging bullet?

Or is vehicle bullet_act doing something?

Reply:
iirc its just 50% to hit someone i can remove if you want this was pasta